### ISSUE TITLE
Wait Support for Websockets

### ISSUE THREAD

**Issue by bitmage** [assaf/zombie#504]
@assaf Continuing our email conversation... Trying to succinctly transcribe here, let me know if anything is misrepresented.  :-)

In my scenario, I initiate a websocket request, and some time later a callback is activated when the response is received.  In the callback a DOM element is inserted, and that's what I would like to wait for before continuing my test.  The problem is that after the websocket request is sent there is no work waiting to be processed by Zombie's event loop, and so the wait function returns prematurely.

I'd like to be able to do:

``` coffee-script
zombie.wait {element: '.chatBox'}, ->
```

So it would be nice to have a flag or an alternate wait method which would only return if:
1. the predicate returns true
2. the timeout is reached

from @assaf:

> That would just add complexity:
> - We went from two states (no events/some events) to three (no events/some events/waiting anyway)
> - Added another option that has to be tested and work in combination with duration, completion function, etc
> - You have to understand the internals to understand why you would use that option
> - And in spite of all of that, web sockets can still fire events _after_ waiting completed

And previously:

> What should happen instead:
> - The test waits long enough for something to happen (e.g. using setTimeout)
> - Some data arrives from the server
> - Websocket client queues event to act on it
> - The test now calls wait(callback)
> - Event loop processes all events
> - wait(callback) gets called
> 
> So two things are missing. On the Zombie side, queuing the web socket events so they only get to execute while the event loop is running (i.e. during a call to wait).
> 
> On the test side, waiting for the server to get data to the client before calling wait(callback).

I don't think this process makes sense, because I still have to guess at how long the network activity will take.  If I give it 200ms, then at that point all the events in question will probably be done.  I might as well just use setTimeout, and not use wait at all.

But I think wait could be useful as I described above.  It's very common that we will be waiting on the network or an external process, and the consumer's job shouldn't be additionally complicated by that.  Returning when the event loop is done seems like an optimization, not a necessity.

Jasmine's waitsFor function is an example of this in practice:

http://pivotal.github.io/jasmine/#section-24

What I'm after is ease of use by the API consumer.  If I'm misunderstanding anything, or there's a better implementation that gets us there, then I'm all ears.

Thanks!
